android/app/src/main/java/com/zulipmobile/MainActivity.kt

```diff
@@ -28,6 +29,45 @@ open class MainActivity : ReactActivity() {
         }
     }
 
+    /* Returns true just if we did handle the intent. */
+    private fun maybeHandleIntent(
+        intent: Intent?,
+        contentResolver: ContentResolver
+    ): Boolean {
+        if (intent == null) {
+            return false
+        }
+
+
+        val host = (application as ReactApplication).reactNativeHost
+        val reactContext = host.tryGetReactInstanceManager()?.currentReactContext
+
+        when (intent.action) {
+            // Share-to-Zulip
+            Intent.ACTION_SEND, Intent.ACTION_SEND_MULTIPLE -> {
+                handleSend(intent, reactContext, contentResolver)
+                return true
+            }
+            // Launch MainActivity on tapping a notification
+            Intent.ACTION_VIEW -> {
+                // The web-auth intent's action is also VIEW; leave that for RN
+                //   to handle.
+                if (!intent.hasExtra(EXTRA_NOTIFICATION_DATA)) {
+                    return false
+                }
+
+                val data = intent.getBundleExtra(EXTRA_NOTIFICATION_DATA)
+                logNotificationData("notif opened", data!!)
```

Comment:
This can be cleaned up slightly by putting the `getBundleExtra` first, and then checking if it's null instead of calling `hasExtra`. Then the `!!` becomes unnecessary.

That also means that if somehow we get an intent with an `EXTRA_NOTIFICATION_DATA` extra that isn't a Bundle, then we don't crash at the `!!` and instead just ignore it here as unhandled.